src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Extensions/SyntaxTokenExtensions.cs

```diff
@@ -210,5 +211,21 @@ public static SyntaxToken KeepCommentsAndAddElasticMarkers(this SyntaxToken toke
             => token
                     .WithTrailingTrivia(token.TrailingTrivia.FilterComments(addElasticMarker: true))
                     .WithLeadingTrivia(token.LeadingTrivia.FilterComments(addElasticMarker: true));
+
+        public static bool IsTypeOfCastContextWhereExpressionIsInNextLine(this SyntaxToken token)
```

Comment:
i'm not sure i understand the name of this method.  `IsTypeOf` reads a bit strange to me.  

oh.  i think i see it now.  because a cast expression has a TypeSyntax it is being cast to you're asking if that's teh node we're on...

the only otehr possibility would be IsInCastExpressionTypeWhereExpression...

that might be a little easier to undertand.  but i don't feel strongly.

Reply:
Sure, fixed